app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::V1::UsersController < ApplicationController
+  def index
+    render json: UserSerializer.new(User.all)
+  end
+
+  def show
+    render json: UserSerializer.new(User.find(params[:id]))
+  end
+
+  def create
+    return render json: Error.missing_params, status: :bad_request if missing_params?
+    return render json: Error.mismatched_passwords, status: :bad_request unless passwords_match?
+    return render json: Error.same_email, status: :bad_request if email_in_use?
+
+    render json: UserSerializer.new(User.create(user_params)), status: :created
+  end
+
+  def update
+    render json: UserSerializer.new(User.update(params[:id], user_params))
+  end
+
+  def destroy
+    user = User.find(params[:id])
+    render json: UserSerializer.new(user.delete)
+  end
+
+  private
+
+  def user_params
+    params.permit(:name, :email, :address, :password)
+  end
+
+  def missing_params?
```

Comment:
To me it makes sense for the controller to now about these methods because these methods deal with checking the params and controllers handle params